### ISSUE TITLE
Remove options "items" & "of" & "refType", change prop option "type" to allow arrow-functions

### ISSUE THREAD

**Issue by hasezoey** [typegoose/typegoose#257]
Remove options `items` and `dim` and replace them with something like nestjs & angular
Example:
currently: `@arrayProp({ items: Number, dim: 3 })`
new way: `@prop({ type: () => Number, dim: 3})`

suggested by @andreialecu on discord

**Comment by hasezoey** [comment]
implementation helper:
```ts
function insertAtDeepest(array: Array<any>, options: any) {
  let deepest: Array<any> = array[0];
  while (Array.isArray(deepest[0])) {
    deepest = deepest[0];
  }
  deepest[1] = Object.assign({}, deepest[1], options); // replaced with lodash merge
}

const theArray = [[[Number]]];
insertAtDeepest(theArray, { someOptions: true });
console.log("theArray", theArray);
```

**Comment by hasezoey** [comment]
closing, because released on beta since `8.0.0-beta.1`
